pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -781,6 +813,14 @@ func (c *Cluster) getActivateOSDInitContainer(configDir, namespace, osdID string
 		Resources:       osdProps.resources,
 	}
 
+	if !osdProps.onPVC() && osdInfo.CVMode == "raw" {
+		container.Command = []string{
+			"/bin/bash",
+			"-c",
+			activateRawOsdOnNodeScript,
```

Comment:
Just curious, what does the log look like in the init container when this script runs?

Reply:
```
> kr logs rook-ceph-osd-0-64697c9dd4-tjff5 -c activate
+ OSD_ID=0
++ mktemp
+ OSD_LIST=/tmp/tmp.pwUVsPGKX5
+ ceph-volume raw list
+ cat /tmp/tmp.pwUVsPGKX5
{
    "4243103f-540f-4041-9f33-a809fd534367": {
        "ceph_fsid": "5f766d2e-6ccb-4046-bfa2-f5e59b27e444",
        "device": "/dev/sda",
        "osd_id": 1,
        "osd_uuid": "4243103f-540f-4041-9f33-a809fd534367",
        "type": "bluestore"
    },
    "872dc671-4dd7-4839-9566-6cdfb0c33dd8": {
        "ceph_fsid": "5f766d2e-6ccb-4046-bfa2-f5e59b27e444",
        "device": "/dev/sdb",
        "osd_id": 2,
        "osd_uuid": "872dc671-4dd7-4839-9566-6cdfb0c33dd8",
        "type": "bluestore"
    },
    "fa5d3aa7-4889-4f7f-8a0c-ac3435e59733": {
        "ceph_fsid": "5f766d2e-6ccb-4046-bfa2-f5e59b27e444",
        "device": "/dev/sdc",
        "osd_id": 0,
        "osd_uuid": "fa5d3aa7-4889-4f7f-8a0c-ac3435e59733",
        "type": "bluestore"
    }
}
++ python3 -c '
import sys, json
for _, info in json.load(sys.stdin).items():
	if info['\''osd_id'\''] == 0:
		print(info['\''device'\''], end='\'''\'')
		sys.exit(0)  # don'\''t keep processing once the disk is found
sys.exit('\''no disk found with OSD ID 0'\'')
'
+ DEVICE=/dev/sdc
+ ceph-volume raw activate --device=/dev/sdc --no-systemd --no-tmpfs
Running command: /usr/bin/chown -R ceph:ceph /var/lib/ceph/osd/ceph-0
Running command: /usr/bin/ceph-bluestore-tool prime-osd-dir --dev /dev/sdc --path /var/lib/ceph/osd/ceph-0 --no-mon-config
Running command: /usr/bin/chown -R ceph:ceph /dev/sdc
Running command: /usr/bin/ln -s /dev/sdc /var/lib/ceph/osd/ceph-0/block
Running command: /usr/bin/chown -R ceph:ceph /var/lib/ceph/osd/ceph-0
--> ceph-volume raw activate successful for osd ID: 0
```

Reply:
RE: https://github.com/rook/rook/pull/7951#discussion_r635629467 You can see the JSON output from the `cat /tmp/tmp....` above.